packages/gatsby/src/bootstrap/get-config-file.js

```diff
@@ -0,0 +1,34 @@
+/* @flow */
+const levenshtein = require('fast-levenshtein');
+const fs = require('fs-extra');
+const testRequireError = require(`../utils/test-require-error`);
+const report = require(`gatsby-cli/lib/reporter`);
+const chalk = require('chalk');
+
+function isNearMatch(fileName: string, configName: string, distance: number): boolean {
+  return levenshtein.get(fileName, configName) <= distance;
+}
+
+module.exports = async function getConfigFile(rootDir: string, configName: string, distance: number = 3) {
+  const configPath = `${rootDir}/${configName}`;
+  try {
+    return require(configPath);
+  } catch (err) {
+    const nearMatch = await fs.readdir(rootDir)
+      .then(files => {
+        return files
+          .find(file => {
+            const fileName = file.split(rootDir).pop();
+            return isNearMatch(fileName, configName, distance);
+          });
+      });
+    if (testRequireError(configPath, err) && nearMatch) {
+      console.log('');
+      report.info(`The file ${chalk.bold(nearMatch)} looks like ${chalk.bold(configName)}, please rename.`);
+      console.log('');
+    }
+
+    report.error(`Could not load ${configName}`, err);
```

Comment:
Sites aren't required to have a `gatsby-config.js` file e.g. see https://github.com/gatsbyjs/gatsby-starter-hello-world

So there's the error path when the config file can't be required. Then there's when there's *not* a gatsby-config.js file but there is another file that's named in a very similar way. I think that should be an error as well since the user almost certainly meant to name the file `gatsby-config.js` but we should print out a different error i.e. don't print the stack trace as that's meaningless in this error path — just the message that they made a typo when naming the config file.

Reply:
TIL. Fixed this in the latest commit.